storage/src/adb/operation/fixed/ordered.rs

```diff
@@ -22,73 +22,81 @@ pub enum Operation<K: Array + Ord, V: CodecFixed> {
 
     /// Indicates all prior operations are no longer subject to rollback, and the floor on inactive
     /// operations has been raised to the wrapped value.
-    CommitFloor(Location),
+    CommitFloor(Option<V>, Location),
 }
 
-impl<K: Array + Ord, V: CodecFixed> Operation<K, V> {
-    // For a compile-time assertion that operation's array size is large enough to handle the commit
-    // operation, which requires 9 bytes.
-    const _MIN_OPERATION_LEN: usize = 9;
-
-    /// Asserts that the size of `Self` is greater than the minimum operation size.
-    #[inline(always)]
-    const fn assert_valid_size() {
-        assert!(
-            Self::SIZE >= Self::_MIN_OPERATION_LEN,
-            "array size too small for commit op"
-        );
+impl<K: Array, V: CodecFixed> Operation<K, V> {
+    // Commit op has a context byte, an option indicator, a metadata value, and a u64 location.
```

Comment:
Nit: I think this is the more commonly used word? I think it should be clear what this is either way.

```suggestion
    // Commit op has a discriminant, an option indicator, a metadata value, and a u64 location.
```